### PR TITLE
[ALLUXIO-2182] Added capacity information per tier

### PR DESCRIPTION
**Pull request by yupeng9**
https://alluxio.atlassian.net/browse/ALLUXIO-2182?jql=text%20~%20%22metrics%22

Part of a series of metrics improvement.
Added the capacity information (total, used, free) per tier for both entire cluster (collected from master) and per worker.

The metrics name is in the pattern of Capacity[Total|Used|Free]Tier(Name). For example, CapacityTotalTierMEM.

### CODE REVIEW

**Comment by yupeng9** [comment]
hmm, i dont think we have any test covering the metrics system.

**Comment by aaudiber** [comment]
It would be valuable to develop test infrastructure for testing metrics, especially now that we are undertaking an effort to overhaul them

**Comment by yupeng9** [comment]
okay, then probably we can take this as a separate task in a future PR

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 1]
```diff
@@ -193,6 +193,7 @@ public Object getFieldValue(MasterWorkerInfo o) {
   DefaultBlockMaster(MasterContext masterContext, Clock clock,
       ExecutorServiceFactory executorServiceFactory) {
     super(masterContext, clock, executorServiceFactory);
+    mGlobalStorageTierAssoc = new MasterStorageTierAssoc();
```
Comment (aaudiber) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: can we make this final now?

**Comment by aaudiber** [comment]
separate PR is fine, but ideally we could merge the infrastructure changes before this PR so that the tests can be included with the code that they test. If we're planning on adding metric test infra as a part of 1.8.0, it's better to do it before the metrics code changes so that we have it available in all the metric change PRs. I'd also like to avoid creating test debt as much as possible.

**Comment by yupeng9** [comment]
I'm fine with that.

**Pull request merged**